File: app-bar-provider/src/main/java/com/sequenia/app_bar_provider/AppBarViews.java
package com.sequenia.app_bar_provider;

import androidx.appcompat.widget.Toolbar;
import android.view.ViewGroup;

import com.google.android.material.appbar.AppBarLayout;
import com.google.android.material.appbar.CollapsingToolbarLayout;

public interface AppBarViews {
    AppBarLayout getAppBar();
    Toolbar getToolbar();
    ViewGroup getCollapsingContent();
    CollapsingToolbarLayout getCollapsingToolbarLayout();
}

File: app-bar-provider/src/main/java/com/sequenia/app_bar_provider/AppBarViews.kt
package com.sequenia.app_bar_provider

import android.view.ViewGroup
import androidx.appcompat.widget.Toolbar
import com.google.android.material.appbar.AppBarLayout
import com.google.android.material.appbar.CollapsingToolbarLayout

interface AppBarViews {
    fun getAppBar(): AppBarLayout?
    fun getToolbar(): Toolbar?
    fun getCollapsingContent(): ViewGroup?
    fun getCollapsingToolbarLayout(): CollapsingToolbarLayout?
}